pkg/validation/app.go

```diff
@@ -397,8 +398,34 @@ func (v *Validator) validateUserConfig(ctx context.Context, cr v1alpha1.App) err
 		// managed by cluster-operator. So we don't need to restrict
 		// the name.
 		if key.CatalogName(cr) == defaultCatalogName && key.AppName(cr) != nginxIngressControllerAppName {
+			// This check is for `cluster-operator` only. For CAPI clusters, that does not rely on
+			// `cluster-operator`, the names could be any, but since it hasn't been conditioned earlier,
+			// making the whole function conditional now, when CAPI is well-established, may have some
+			// hard-to-predict consequences. Hence it is safer to let the function operate as usual,
+			// but in addition put some extra conditions like below.
+
+			// User ConfigMap name must, in general, match the '<App_CR_name>-user-values',
+			// with one exception below.
 			configMapName := fmt.Sprintf("%s-user-values", cr.Name)
-			if key.UserConfigMapName(cr) != configMapName {
+
+			nameMismatch := key.UserConfigMapName(cr) != configMapName
+
+			// As the result of misconfiguration for the legacy clusters, see
```

Comment:
As per the original issue this does not look like a misconfiguration but the only way forward :D

Reply:
@QuentinBisson You are right, somehow I forgot we are talking about in-cluster apps here, my comment in the issue is the result of this as well, heh 😞 In case of in-cluster apps I think we told people ourselves at some point to add prefix or suffix to these apps as they are installed in the MC itself. I will fix the description.